packages/rancher-vsphere-csi/charts/questions.yaml

```diff
@@ -5,46 +5,83 @@ questions:
     type: boolean
     default: true
     required: true
-    group: Configuration
+    group: vCenter Configuration
     show_subquestion_if: true
     subquestions:
       - variable: vCenter.host
         label: vCenter Host
         description: IP address or FQDN of the vCenter
         type: string
-        group: Configuration
 
       - variable: vCenter.datacenters
         description: Comma-separated list of paths to data centers. E.g "<dc1-path>, <dc2-path>, ..."
         label: Data Centers
         type: string
-        group: Configuration
 
       - variable: vCenter.username
         label: Username
         description: Username for vCenter
         type: string
-        group: Configuration
 
       - variable: vCenter.password
         label: Password
         description: Password for vCenter
         type: password
-        group: Configuration
 
   - variable: vCenter.configSecret.name
     label: CSI vSphere Config Secret Name
     description: Name of the Secret that contains a CSI vSphere config and credentials (Will not be visible in the API. More info in the README)
     type: string
-    group: Configuration
+    group: vCenter Configuration
```

Comment:
The overall PR looks good to me, but I'm a little concerned about the fact that we are doing a heavy refactor of the chart within the 2.5 release line of Rancher, which will presumably be forward-ported only in 2.6.2's release later this year.

Specifically, here is the upgrade scenario that concerns me:
- User is currently on Rancher 2.5 using the older version of the vsphere chart
- User upgrades to Rancher 2.5.10
- User upgrades to latest vsphere chart version 2.3.0
- User upgrades Rancher to 2.6.1

In this situation, will the user be able to upgrade from `2.3.0` -> `100.0.0`? I presume the answer here is no, since that will cause them to downgrade vsphere-cpi versions.

In addition, if a user chooses to keep `2.3.0` in Rancher 2.6 since there is no path for upgrade, is there a way to do an in-place Helm upgrade / re-install after migrating? The dev-v2.5 ChartRepository will no longer exist, so I do not think this is an option either.

Reply:
From what I can tell, this probably would merit an OOB release for vsphere after 2.5.10 is released into the 2.6 charts release line to ensure that this type of upgrade path is still supported for users to upgrade to 2.6.1.